### ISSUE TITLE
warning: 'out' may be used uninitialized [-Wmaybe-uninitialized] in AVX=512z builds with gcc12

### ISSUE THREAD

**Issue by valassi** [madgraph5/madgraph4gpu#594]
There is a warning: 'out' may be used uninitialized [-Wmaybe-uninitialized] in AVX=512z builds with gcc12
```
ccache /cvmfs/sft.cern.ch/lcg/releases/gcc/12.1.0-57c96/x86_64-centos9/bin/g++  -O3  -std=c++17 -I. -I../../src -I../../../../../tools -I../../../../../test/googletest/googletest/include -I../../../../../test/googletest/googletest/include -DUSE_NVTX -Wall -Wshadow -Wextra -ffast-math  -fopenmp -march=skylake-avx512 -DMGONGPU_PVW512  -DMGONGPU_FPTYPE_DOUBLE -DMGONGPU_FPTYPE2_DOUBLE -I/usr/local/cuda-12.0/include/ -fPIC -c testxxx.cc -o testxxx.o
In file included from CPPProcess.h:13,
                 from testxxx.cc:3:
In function 'mgOnGpu::fptype_v fpsqrt(const mgOnGpu::fptype_v&)',
    inlined from 'void mg5amcCpu::ixzxxx(const mgOnGpu::fptype*, int, int, mgOnGpu::fptype*, int) [with M_ACCESS = KernelAccessMomenta<false>; W_ACCESS = KernelAccessWavefunctions<false>]' at ../../src/HelAmps_sm.h:384:36,
    inlined from 'virtual void SIGMA_SM_GG_TTX_CPU_XXX_testxxx_Test::TestBody()' at testxxx.cc:265:59:
../../src/mgOnGpuVectors.h:209:43: warning: 'out' may be used uninitialized [-Wmaybe-uninitialized]
  209 |   for( int i = 0; i < neppV; i++ ) out[i] = fpsqrt( v[i] );
      |                                    ~~~~~~~^~~~~~~~~~~~~~~~
../../src/mgOnGpuVectors.h: In member function 'virtual void SIGMA_SM_GG_TTX_CPU_XXX_testxxx_Test::TestBody()':
../../src/mgOnGpuVectors.h:208:12: note: 'out' was declared here
  208 |   fptype_v out;
      |            ^~~

```

**Comment by valassi** [comment]
This will be fixed in MR #594, I am closing this

**Comment by valassi** [comment]
NOTE: this introduces an extra initialisation. This changes the binary also for gcc11.3, I do see a small change in the objdump output
https://github.com/madgraph5/madgraph4gpu/pull/595/commits/6d0b2df71e958daa2c8fc1bf7f09c862282b90ed
For 512z was
```
=Symbols in CPPProcess.o= (~sse4: 0) (avx2: 1381) (512y: 69) (512z: 9966)
```
Is now
```
=Symbols in CPPProcess.o= (~sse4: 0) (avx2: 1383) (512y: 67) (512z: 9966)
```

I will assume however that this is a negligible performance impact
